docs/modules/angular/pages/cookbooks/vanilla-to-onecx/migrate-to-onecx.adoc

```diff
@@ -0,0 +1,279 @@
+= Migrate vanilla Angular apps to OneCX
+Bastian Jakobi bastian.jakobi@capgemini.com
+:source-highlighter: highlight.js
+
+The goal of this guide is to provide step by step instructions on how to migrate a vanilla Angular app (e.g. generated with Angular CLI) to a OneCX compatible microfrontend application. 
+
+== General
+=== Prerequisites
+This guide assumes that an Angular project is already set up and that a OneCX platform instance is up and running. At the time of writing, some required dependencies are only compatible with Angular version 18. Therefore the application must be updated to Angular 18 before migrating. Additionally, the Angular application should be based on feature modules, which are lazy loaded before routing.
+
+Note that this guide does not cover general Angular project setup or the steps to configure OneCX.
+
+== Migration steps
+The goal of this migration is to transform a "vanilla" Angular app into a OneCX compatible app that exposes a microfrontend which can be loaded and displayed by the OneCX platform. To achieve this goal, a few migration steps have to be completed. All steps have been tested in a fresh app generated with Angular CLI but should also work in other Angular apps (e.g. apps generated with NX).
+
+=== Ensure that your Angular app runs in standalone mode
+Before starting the actual app migration, the correct functionality of the application in its "pre-migration" state should be verified by running existing tests, building the app and running it in a browser. This way it can be ensured that the app was working as expected before the migration, which will help to prevent unexpected errors in later stages of the migration.
+
+=== Install necessary dependencies
+In order to turn an app into a OneCX compatible microfrontend it's necessary to first install some dependencies which allow wrapping the app inside a OneCX compatible webcomponent and exposing it as a remote entrypoint.
+
+.Dependencies
+[source,console]
+----
+npm i @angular/elements @angular-architects/module-federation @onecx/angular-integration-interface @onecx/angular-webcomponents @webcomponents/webcomponentsjs
+----
+To resolve potential peer-dependency errors during the installation of `@angular/elements`, the dependency can be manually added to `package.json`, using the version number used for other `@angular/` packages in the project.
+
+.Dev-Dependencies
+[source,console]
+----
+npm i -D webpack webpack-merge ngx-build-plus
+----
+
+=== Refactor project configuration to use ngx-build-plus builders
+To correctly bundle an app for usage in a microfrontend context the default Angular bundler has to be replaced with ngx-build-plus and a few additional configuration options have to be specified.
+
+Specifically the projects configuration file (e.g. `angular.json`) has to be modified in the following way:
```

Comment:
Please also add instructions for nx apps

Reply:
**Note to myself: Test guide with plain nx app and adjust instructions if needed**